library/declare.ml

```diff
@@ -44,7 +44,9 @@ let if_xml f x = if !Flags.xml_export then f x else ()
 
 type section_variable_entry =
   | SectionLocalDef of Safe_typing.private_constants definition_entry
-  | SectionLocalAssum of types Univ.in_universe_context_set * polymorphic * bool (** Implicit status *)
+  | SectionLocalAssum of { type_context : types Univ.in_universe_context_set;
```

Comment:
The "forward compatible" way of writing that code is
```
type a = Foo of foo
and foo = { ... }
```
Then in OCaml 4.03 you remove the intermediate `foo`.